templates/vue/src/store/actions.js

```diff
@@ -405,68 +405,82 @@ export function setTapestryErrorReporting({ commit }, isEnabled) {
   commit("setTapestryErrorReporting", isEnabled)
 }
 
-export function resetNodeNavigation({ commit }, nodeId) {
+export function resetNodeNavigation({ commit, getters }, nodeId) {
   commit("setNodeNavigation", {
+    ...getters.getLinkNavigation(nodeId),
     stack: [nodeId],
-    siblings: [],
-    siblingPosition: -1,
     linkMode: false,
   })
 }
 
-export function goToNodeChildren({ commit, state, getters }) {
-  if (state.nodeNavigation.stack.length > 0) {
-    const nodeId = state.nodeNavigation.stack[state.nodeNavigation.stack.length - 1]
-    const childrenIds = getters.getDirectChildren(nodeId)
-    if (childrenIds.length > 0) {
-      commit("setNodeNavigation", {
-        stack: [...state.nodeNavigation.stack, childrenIds[0]],
-        siblings: childrenIds,
-        siblingPosition: 0,
-      })
-      return childrenIds[0]
-    }
+export function setNavLinkMode({ commit, state }, linkMode) {
+  if (linkMode && state.nodeNavigation.currentLinked.type === null) {
+    return false
   }
-  return false
+  commit("setNodeNavigation", {
+    linkMode: linkMode,
+  })
+  return true
 }
```

Comment:
Would you be able to explain why this function has to return true or false? I see the return value is not being used in our code so far.

Reply:
Indeed the result is not being used as of now. I just put the return value here in case we want to inform the user that they cannot enter link mode when there are no links attached to the node. If we don't need that functionality then I can remove the return value from this function without any issues.

Reply:
Got it, thanks! Since this PR is not ready for final review yet, I think there's no harm in keeping it, in case it will be used in the future.